_posts/2020-03-18-http-security-headers.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: HTTP Security Headers
+author: massimo
+excerpt: >
+  Your fancy front-end framework won't print unescaped HTML. But what if it does after the next update? Well, that's why you have a WAF. What if there's a way to circumvent it? There's always a what if. This is not a comprehensive list of all security headers. These are the bare minimum every non-trivial site should use.
+tags:
+- infosec
+- software security
+- devsecops
+---
+
+Defense in depth as defined by Wikipedia is a:
+> concept in which multiple layers of security controls (defense) are placed throughout an information technology (IT) system
+
+## Onion in eye
+
+![onion](/img/http-security-headers/onion.jpg)
+
+The purpose of aforementioned security controls is to provide defense against attacks in a layered fashion. If one security control fails, there is another lower in the stack. This deters the attacker and gives the defender more time to contain the threat before the next control is breached. Like when an onion is peeled, every layer brings more tears.
```

Comment:
deeper in the stack? Lower implies somehow less important, deeper might work better with onion comparison.,

Reply:
Fixed